src/nanotron/trainer.py

```diff
@@ -215,7 +215,7 @@ def __init__(
             )
 
         # Define iteration start state
-        if self.init_checkpoint_path is not None:
+        if self.init_checkpoint_path is not None and self.config.checkpoints.load_lr_scheduler:
```

Comment:
is this only lr related?

Reply:
I think when you want to do a new lr schedule, you also want to do a different data stages + reset the "last_train_step" argument